packages/reports/src/types.ts

```diff
@@ -9,19 +9,38 @@ import { AccessPolicy } from '@tupaia/access-policy';
 import { Authenticator } from '@tupaia/auth';
 import { TupaiaDatabase, ModelRegistry } from '@tupaia/database';
 
+export interface FetchReportQuery extends Query {
+  organisationUnitCode: string;
+  period?: string;
+}
+
+export interface FetchReportParams extends ParamsDictionary {
+  reportCode: string;
+}
+
+interface ReportsRequestBody {
+  emailAddress: string;
+  password: string;
+}
+
 export interface ReportsRequest<
-  P = ParamsDictionary,
-  ResBody = any,
+  P = FetchReportParams,
+  ResBody = unknown,
   ReqBody = ReportsRequestBody,
-  ReqQuery = Query
+  ReqQuery = FetchReportQuery
 > extends Request<P, ResBody, ReqBody, ReqQuery> {
   accessPolicy: AccessPolicy;
   authenticator: Authenticator;
   database: TupaiaDatabase;
   models: ModelRegistry;
 }
 
-interface ReportsRequestBody {
-  emailAddress: string;
-  password: string;
+export interface Event {
+  event: string;
+  eventDate: string;
+  orgUnitName: string;
+  orgUnit: string;
+  dataValues?: {
+    [key: string]: string | number | boolean;
```

Comment:
This is my only suggestion... I believe we don't cast event data values to `boolean`, but I'm not 100% sure

```suggestion
    [key: string]: string | number;
```

Reply:
Ahh true... boolean values are regarded as numbers huh? I'm happy to change this 👍